4.1 AI agent chat (#670)
SimpleDataLabsInc/prophecy-docs: docs/analysts/development/ai-agent/ai-transform.md

---
title: Transform data
id: ai-transform
slug: /analysts/ai-transform
description: Transform data sources using Prophecy's AI agent
tags:
  - analyst
  - ai
  - agent
---

You can use Prophecy's AI agent to generate transformations based on natural language prompts. When you describe a data operation, the agent adds the corresponding gems to the pipeline and provides a summary of the changes in the chat interface.

The following sections describe how to add transformations, review modifications, and restore previous versions of the pipeline.

:::info Prerequisite
You need at least one [Source gem](/analysts/source-target) in your pipeline to add transformation gems with AI chat.

> :::

Reviewer: Suggestion: I've been actually putting in a heading for pre-requisites in Transpiler docs. I get that this was only 1 item, but I think it's still better as a section VS in a box here. I think a section makes it more noticable.

## Provide a transformation

> To generate a transformation, enter a prompt that describes the desired data operation. The agent returns:

Reviewer: ```suggestion
To generate a transformation, enter a prompt that describes the desired data operation.

The agent returns:
```

- One or more gems on the pipeline canvas
- A description of the applied changes
- SQL execution logs

![Agent SQL logs](img/agent-sql-logs.png)

### Project history

All changes made by the agent are saved in the project [version history](/analysts/versioning). Commits will be clearly marked as authored by the agent.

:::note
If you did not save your project before interacting with the AI agent, Prophecy will automatically save your changes before the agent proceeds.
:::

## Inspect pipeline changes

To understand the agent changes:

> 1. Select **Inspect** in the chat next to a transformation.

Reviewer: ```suggestion
1. Select **Inspect** at the bottom of the chat for a transformation.
```

Nit: To be super specific -- this also answers the question I had about the Preview thing to click above -- so maybe clarifying that too on the previous page, AND/OR update the screenshot there to show it if possible.

1. Review the configuration panel beginning with the first modified gem (highlighted in yellow to indicate modification).
1. Use the **Previous** and **Next** controls to move through other modified gems in sequence.
1. Hover over the **Previous** and **Next** button to display a minimap of the pipeline. This shows you the specific gem you are viewing in the context of the pipeline.
1. Examine both the input and output of each gem to confirm that the transformation produces the expected result.

## Restore a previous state of the pipeline

To revert changes or try another transformation from a previous state:

> 1. Select **Restore** from a reply in the chat history

Reviewer: ```suggestion
1. Select **Restore** from the reply you want to revert to in the chat history.
```

Nit: to be super specific

> 1. The pipeline will match the earlier version.

Reviewer: Question: Should this be a step? I feel like this is more of a note/sentence.

:::note
You can also manage versions from the main project [version history](/analysts/versioning).
:::

## Create output tables

After adding various data transformations in your pipeline:

1. Ask the agent to save the result as a table.

> 1. The agent writes the data to the default database and schema in your connected fabric.

Reviewer: Question: Can users specify where they want the agent to write their data to? (like to override the default)

Author: Nope 😅
